src/player_display.cpp

```diff
@@ -765,460 +878,560 @@ Strength - 4;    Dexterity - 4;    Intelligence - 4;    Perception - 4" ) );
+        mvwprintz( w_info, 3, 21, c_magenta, "%3d%%", player.read_speed( false ) );
+        mvwprintz( w_info, 4, 1, c_magenta, _( "Skill rust:" ) );
+        mvwprintz( w_info, 4, 22, c_magenta, "%2d%%", player.rust_rate( false ) );
+        mvwprintz( w_info, 5, 1, c_magenta, _( "Crafting bonus:" ) );
+        mvwprintz( w_info, 5, 22, c_magenta, "%2d%%", player.get_int() );
+    } else if( line == 3 ) {
+        // Display information on player perception in appropriate window
+        mvwprintz( w_stats, 5, 1, h_light_gray, _( "Perception:" ) );
+        fold_and_print( w_info, 0, 1, FULL_SCREEN_WIDTH - 2, c_magenta,
+                        _( "Perception is the most important stat for ranged combat.  It's also used for "
+                           "detecting traps and other things of interest." ) );
+        mvwprintz( w_info, 4, 1, c_magenta, _( "Trap detection level:" ) );
+        mvwprintz( w_info, 4, 23, c_magenta, "%2d", player.get_per() );
+        if( player.ranged_per_mod() > 0 ) {
+            mvwprintz( w_info, 5, 1, c_magenta, _( "Aiming penalty:" ) );
+            mvwprintz( w_info, 5, 21, c_magenta, "%+4d", -player.ranged_per_mod() );
+        }
+    }
+    wrefresh( w_stats );
+    wrefresh( w_info );
+
+    std::string action = ctxt.handle_input();
+    if( action == "DOWN" ) {
+        line++;
+        if( line >= 4 ) {
+            line = 0;
+        }
+    } else if( action == "UP" ) {
+        if( line <= 0 ) {
+            line = 3;
+        } else {
+            line--;
+        }
+    } else if( action == "PREV_TAB" ) {
+        draw_inactive_stats();
+        line = 0;
+        decrement_tab();
+    } else if( action == "NEXT_TAB" ) {
+        draw_inactive_stats();
+        line = 0;
+        increment_tab();
+    } else if( action == "QUIT" ) {
+        this->done = true;
+    }
+    mvwprintz( w_stats, 2, 1, c_light_gray, _( "Strength:" ) );
+    mvwprintz( w_stats, 3, 1, c_light_gray, _( "Dexterity:" ) );
+    mvwprintz( w_stats, 4, 1, c_light_gray, _( "Intelligence:" ) );
+    mvwprintz( w_stats, 5, 1, c_light_gray, _( "Perception:" ) );
+    wrefresh( w_stats );
+}
+
+void player_display_info::draw_inactive_stats()
+{
+    mvwprintz( w_stats, 0, 0, c_light_gray, header_spaces.c_str() );
+    center_print( w_stats, 0, c_light_gray, title_STATS );
+    wrefresh( w_stats );
+}
+
+void player_display_info::draw_active_encumbrance(player& player, input_context &ctxt)
+{
+
+    werase( w_encumb );
+    center_print( w_encumb, 0, h_light_gray, title_ENCUMB );
+    player.print_encumbrance( w_encumb, line );
+    wrefresh( w_encumb );
+
+    werase( w_info );
+    std::string s;
+
+    body_part bp = line <= 11 ? all_body_parts[line] : num_bp;
+    bool combined_here = ( bp_aiOther[line] == line + 1 ||
+                           bp_aiOther[line] == line - 1 ) && // first of a pair
+        should_combine_bps( player, line, bp_aiOther[line] );
+    s += get_encumbrance_description( player, bp, combined_here );
```

Comment:
`s` is only ever written here. It can be made defined and initialized here and const.